src/Build/Instance/HostObjectException.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Build.Execution
+{
+    [Serializable]
+    internal sealed class HostObjectException : Exception
+    {
+        private const string ErrorMessagePrefix = "Error for HostObject:";
```

Comment:
@rainersigwald should we assign a error code for this? And how?

Reply:
Is it a user-facing error, or one that we believe will only ever be because of a bug in VS/MSBuild? If the former, I put a doc up in https://github.com/Microsoft/msbuild/pull/4264 -- please leave comments if it's unclear.

Reply:
It is later. Only VS or MSBuild dev should see this. If the user see this, it is a bug

Reply:
Then I'd probably derive this from `InternalErrorException` and **not** assign it a code.

Reply:
@wli3 pointed out offline that `InternalErrorException` a) is `sealed` and b) doesn't really convey more information here, so I agree that we should leave it as is.